modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -3254,19 +3271,73 @@ public LocalSnapshotSender(String snpName, @Nullable String snpPath) {
 
                 assert totalBytes % pageSize == 0 : "Given file with delta pages has incorrect size: " + fileIo.size();
 
+                int pagesCnt = (int)(totalBytes / pageSize);
+
+                Integer batchSize = deltaSortBatch.get();
+
+                GridIntIterator iter;
+
+                if (batchSize == null || batchSize <= 0)
+                    iter = U.forRange(0, pagesCnt);
+                else {
+                    iter = new GridIntIterator() {
+                        private int idx = 0;
+
+                        private Iterator<Integer> sortedIter;
+
+                        @Override public boolean hasNext() {
+                            if (sortedIter == null || !sortedIter.hasNext()) {
+                                try {
+                                    advance();
+                                }
+                                catch (Exception e) {
+                                    throw new IgniteException(e);
+                                }
+                            }
+
+                            return sortedIter.hasNext();
+                        }
+
+                        @Override public int next() {
+                            return sortedIter.next();
```

Comment:
It will produce NullPointerExcepction here if we call the `next` first time. It seems it should be IllegalStateException or something similar.

Reply:
Added check.